wallet/test/mock_remote.go

```diff
@@ -87,77 +89,154 @@ func initializeRandomValues(r *MockRemote, rng *rand.Rand) {
-	m.callSign = f
+	r.MockChannelState = test.MakeRandomChannelState(rng)
 }
 
-func makeCallSignDefault(r *MockRemote) func(request wire.SigningRequest) (wire.SigningResponse, error) {
-	return func(request wire.SigningRequest) (wire.SigningResponse, error) {
-		reqAddr, err := request.PubKey.Decode()
-		if err != nil {
-			return wire.SigningResponse{}, fmt.Errorf("unable to decode PubKey from request")
-		}
-		if !reqAddr.Equal(&r.MockAddress) {
-			return wire.SigningResponse{}, fmt.Errorf("invalid public key for mock remote")
-		}
-
-		if request.Message != r.MockMessageString {
-			return wire.SigningResponse{}, fmt.Errorf("invalid data for mock remote")
-		}
-		return wire.Signature{Signature: r.MockSignatureString}, nil
-	}
+func (m *MockRemote) SetCallEndpoint(f func(string, interface{}, interface{}) error) {
+	m.callEndpoint = f
 }
 
-func makeCallVerifyDefault(r *MockRemote) func(wire.VerificationRequest) (wire.VerificationResponse, error) {
-	return func(request wire.VerificationRequest) (wire.VerificationResponse, error) {
-		reqAddr, err := request.PubKey.Decode()
-		if err != nil {
-			return false, fmt.Errorf("unable to decode PubKey from request")
-		}
-		if !reqAddr.Equal(&r.MockAddress) && !reqAddr.Equal(&r.UnavailableAddress) {
-			return false, fmt.Errorf("invalid public key for mock remote")
-		}
-		if reqAddr.Equal(&r.UnavailableAddress) {
-			return false, nil
-		}
-
-		if request.Message != r.MockMessageString {
-			return false, fmt.Errorf("invalid data for mock remote")
-		}
-		if request.Signature.Signature == r.MockSignatureString {
-			return true, nil
-		}
-		if request.Signature.Signature == r.OtherSignatureString {
-			return false, nil
-		}
-		if request.Signature.Signature == hex.EncodeToString(r.InvalidSignatureShorter) ||
-			request.Signature.Signature == hex.EncodeToString(r.InvalidSignatureLonger) {
-			panic("mock remote received signature of invalid length to verify")
-		}
-		return false, fmt.Errorf("invalid signature for mock remote")
-	}
-}
+func makeCallEndpointDefault(r *MockRemote) func(string, interface{}, interface{}) error {
+	return func(endpoint string, req interface{}, resp interface{}) error {
+		switch endpoint {
+		case wallet.EndpointSignData:
+			request, ok := req.(wire.SigningRequest)
+			if !ok {
+				return fmt.Errorf("unable to cast request to SingingRequest")
+			}
+			response, ok := resp.(*wire.SigningResponse)
+			if !ok {
+				return fmt.Errorf("unable to cast resp to SingingResponse")
+			}
+			reqAddr, err := request.PubKey.Decode()
+			if err != nil {
+				return fmt.Errorf("unable to decode PubKey from request")
+			}
+			if !reqAddr.Equal(&r.MockAddress) {
+				return fmt.Errorf("invalid public key for mock remote")
+			}
+
+			if request.Message != r.MockMessageString {
+				return fmt.Errorf("invalid data for mock remote")
+			}
+			response.Signature = r.MockSignatureString
+			return nil
```

Comment:
I would prefer explicit implementation for each case, s.t. this function does not grow longer than 200 lines :clown_face: